### PR TITLE
Improve eslint config

### PR DESCRIPTION
**Pull request by amio**
- Get `.ts` files covered 🎉
- Update eslint related dependencies
- Add typescript-eslint [recommended rules](https://github.com/typescript-eslint/typescript-eslint/blob/3e26ab684ae5642e9c50940bedd155e32e189900/packages/eslint-plugin/src/configs/recommended.json), but only enabled ones we already followed, commented ones causing errors.

Commented rules including:

```yaml
  # '@typescript-eslint/no-unused-vars': 1
  # '@typescript-eslint/indent': ['error', 2]
  # '@typescript-eslint/array-type': error
  # '@typescript-eslint/ban-types': error
  # '@typescript-eslint/explicit-member-accessibility': error
  # '@typescript-eslint/member-delimiter-style': error
  # '@typescript-eslint/no-angle-bracket-type-assertion': error
  # '@typescript-eslint/no-explicit-any': warn
  # '@typescript-eslint/no-object-literal-type-assertion': error
  # '@typescript-eslint/no-use-before-define': error
  # '@typescript-eslint/no-var-requires': error
  # '@typescript-eslint/prefer-interface': error
```

### TODO

Go through these commented rules one by one, enable & fix for it or confirm it's disabled. Since enabling some rules would cause massive code change (like `'@typescript-eslint/indent': ['error', 2]`), which might conflict with our ongoing development, we better do this in separated PRs.

### CODE REVIEW

**Review by styfle — APPROVED** [review]
Looks ok. Let's get someone else to approve too 👍

**Comment by amio** [comment]
Ping @leo @TooTallNate @javivelasco, could you help review this? 🙏

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
```
Comment (leo) [inline .eslintrc.yml]: This is a Node (JavaScript) project, so let's use JSON for this.

**Review by leo — CHANGES_REQUESTED** [review]

File: .eslintrc.yml (hunk 1 above)
Comment (amio) [reply .eslintrc.yml]: Sure, but lets use `.eslintrc.js` for including comments 😉

**Review by leo — APPROVED** [review]

**Pull request merged**